CONTRIBUTING.rst

```diff
@@ -320,69 +322,165 @@ The build status for pyiron and all sub packages are given below
+* black.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/black.yml
+* codeql-analysis.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/codeql-analysis.yml
+* contrib-compat.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/contrib-compat.yml
+* coverage.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/coverage.yml
+* deploy.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/deploy.yml
+* docs.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/docs.yml
+* notebooks.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/notebooks.yml
+* pypicheck.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/pypicheck.yml
+* unittests.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/unittests.yml
+
+**UpdateDependabotPR**
+
+This workflow is used to update the dependencies of the repository. 
+Therefore, it retrieves the head of the branch and runs the ./.ci_support/update_environment.py script. 
+The script basically checks the title of the PR and bumps the version number from a package to a new version number.
+After that, the version is written into the environment.yml file located at ./.ci_support/environment.yml.
+If this is done successfully, the workflow saves the changes and pushes back to the branch of the PR.
+
+**atomistic-compat.yml**
+
+This workflow is used to check the compatibility with the pyiron_atomistics. 
+First, the workflow installs PyYAML and clones pyiron_atomistics. 
+After that, the dependencies from pyiron_atomistics/.ci_support/environment.yml are copied into the pyiron_atomistics/environment.yml. 
+Then, the script condamerge.py (./ci_support/condamerge.py) is triggered, which merges the dependencies from pyiron_base/.ci_support/environment.yml into the pyiron_atomistics/environmnet.yml.
+After the setup is done, the script pyironconfig.py runs iand creates the pyiron configuration.
+Following, both for pyiron_atomisstics and pyiron_base a dependency installation is run.
+The last step is running the unittests inside pyiron_atomistics.
+
+**backwards.yml**
+
+This workflow is used to check backwards compatibility. 
+Therefore, the current branch is retrieved, the environment is setup and configured. 
+After that, the script ./ci_support/test_backwards.sh is triggered. 
+This script runs different tests vor different versions of pyiron_base.
+
+**benchmarks.yml**
+
+This workflow is used to run unittests on the current branch. 
+First, the head of the branch is cloned and a conda environment is created. 
+After that, the pyiron configuration is setup and dependencies are installed. 
+Following, the unittests are triggered and run.
+
+**black.yml**
+
+This workflow is used to check the differences and find conflict markers or whitespace errors.
```

Comment:
This is pretty new and not yet documented at all, however, we stared to format our code using the black style https://black.readthedocs.io/en/stable/. This workflow checks if the new code is consistent to the black standard.